app/validators/url_validator.rb

```diff
@@ -1,8 +1,14 @@
 class UrlValidator < ActiveModel::EachValidator
   # The URL validation rules are somewhat overly strict, but should serve;
   # the idea is to prevent attackers from inserting redirecting URLs
-  # that can sometimes be used to attack (e.g., "?...", or ones with <).
-  URL_REGEX = /#{URI.regexp(%w(http https))}/
+  # that can sometimes be used to attack third-party sites via the BadgeApp.
+  # For example, this does *not* includg "?...", "%..", or ones with <).
+  # We can gradually loosen this if needed.  Note that this pattern is
+  # more restrictive than the ones we allow in markdown justifications,
+  # because the BadgeApp doies *not* follow those URLs for further processing.
+  URL_REGEX =
+    %r{\A(|https?://[A-Za-z0-9][-A-Za-z0-9_.]*(/[-A-Za-z0-9_.:/+!,#]*)?)\z}
```

Comment:
Do you want to site a source for this regex in the comment or is it of your own creation?